xml/hwbp_registers.xml

```diff
@@ -95,11 +95,25 @@
         <field name="data" bits="XLEN-1:0" access="R/W" reset="Preset" />
     </register>
 
-    <register name="Trigger Data 3" short="tdata3" address="0x7a3">
-        Trigger-specific data.
+    <register name="Trigger Extra" short="textra" address="0x7a3">
+        Any bit in this register may be hardwired to 0 if its functionality is
+        not implemented. The checks specified in this register apply regardless
+        of \Ftype in \Rtdataone.
+
+        \begin{commentary}
+            TASIDLEN is limited in size when XLEN is less than DXLEN to
+            preserve the maximum number of bits for future use.
+        \end{commentary}
+        <field name="0" bits="XLEN-1:1+TASIDLEN" access="R" reset="0" />
+        <field name="tasid" bits="TASIDLEN:1" access="R/W" reset="0">
+            If XLEN is 32 then TASIDLEN is 9. Otherwise TASIDLEN is 16.
```

Comment:
How does this statement relate to LOG(ASIDMAX) and the above statement that "any of the bits can be hardwired to 0". I *think* you mean that even if TASIDLEN is 16, this entire field may not be writable, if LOG(ASIDMAX) < TASIDLEN. Confirming that's what you mean, and thinking if this needs re-wording if that is the case. You could just say `TASIDLEN` is CEIL(LOG2(ASIDMAX)) if that's what you really mean.

Reply:
I guess let's say you had 7 possible ASID in a system (ASIDMAX = 3). Could I have only 1 writable bit in this field?

Again I think the answer is "yes" just sort of confusing how the different values are defined. The way the priv spec defines it is very clear.

Reply:
I was confusing myself, ASIDMAX is already the width. Will update my comments above.

Reply:
What I'm trying to resolve is a 64-bit core, with the current XLEN being 32. Such a core could have ASIDLEN=16, but even if that is the case, when the current XLEN is 32 I still want TASIDLEN to be 9. This leaves us the most number of bits available for future use in this register, and also makes it easier to use. The running code doesn't have to know the actual max XLEN. Just based on the current XLEN it knows TASIDLEN.

Clearly I completely failed to communicate that. Does this comment make more sense?

Just above I mention explicitly that any bit in the register may be tied to 0 by an implementation, which should cover your case of implementing fewer bits when ASIDLEN<ASIDMAX.